### PR TITLE
feat: writeFloat for seeding performance in SeedingPerformanceWriter.cpp

### PR DESCRIPTION
**Pull request by LuisFelipeCoelho**
Store (floats) total efficiency, fake rate, duplicate rate and purity in output root file as it is done with ckf performance writer.
This is useful for using the seeding parameters in optuna automatic tuning.

### CODE REVIEW

File: Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp [hunk 1]
```diff
@@ -95,10 +97,20 @@ ActsExamples::ProcessCode ActsExamples::SeedingPerformanceWriter::finalize() {
       "/ nMatchedParticles) = "
       << aveNDuplicatedSeeds);
 
+  auto write_float = [&](float f, const char* name) {
+    TVectorF v(1);
+    v[0] = f;
```
Comment (felix-russo) [inline Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp:102]: Does this not work with `std::vector<float>`?

File: Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp [hunk 2]
```diff
@@ -95,10 +97,20 @@ ActsExamples::ProcessCode ActsExamples::SeedingPerformanceWriter::finalize() {
...
+    m_outputFile->WriteObject(&v, name);
+  };
+
   if (m_outputFile != nullptr) {
     m_outputFile->cd();
     m_effPlotTool.write(m_effPlotCache);
     m_duplicationPlotTool.write(m_duplicationPlotCache);
+    write_float(eff, "eff_seeds");
+    write_float(fakeRate, "fakerate_seeds");
+    write_float(duplicationRate, "duplicaterate_seeds");
```
Comment (felix-russo) [inline Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp:112]: Maybe add underscores for readability?

File: Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp (hunk 1 above)
Comment (LuisFelipeCoelho) [reply Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp:102]: yes, it stores a `std::vector<float>`

File: Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp (hunk 2 above)
Comment (LuisFelipeCoelho) [reply Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp:112]: I can add underscores if you want, but I just copied it from the ckf performance writer. 
For consistency I think I would have to change there as well.

Comment (AJPfleger) [reply Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp:112]: shouldn't the function itself not be named `writeFloat` or do we have a different convention for lambdas?
@paulgessinger

File: Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp (hunk 1 above)
Comment (felix-russo) [reply Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp:102]: I meant, can't you just do 
```suggestion
    std::vector<float> v {f};
```

File: Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp (hunk 2 above)
Comment (felix-russo) [reply Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp:112]: Ah, I meant in the string not in the function name ^^

File: Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp (hunk 1 above)
Comment (felix-russo) [reply Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp]: If this works, you could maybe remove the includes!

File: Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp [hunk 3]
```diff
@@ -22,6 +22,8 @@
 #include <vector>
 
 #include <TFile.h>
+#include <TVectorFfwd.h>
+#include <TVectorT.h>
```
Comment (paulgessinger) [inline Examples/Io/Performance/ActsExamples/Io/Performance/SeedingPerformanceWriter.cpp:26]:
```suggestion
```

**Review by felix-russo — DISMISSED** [review]

**Review by paulgessinger — DISMISSED** [review]

**Comment by paulgessinger** [comment]
- Do we understand why the physmon outputs change? This should not be affected by some extra variables being written to the output files.
- The formatting seems to need a fix
- This PR picked up conflicts.
- The `vector<float>` output breaks uproot reading, and therefore unfortunately breaks the root file hash comparison. Not sure what the best fix for this is. It might be possible to skip these fields when calculating the hashes, but not entirely sure.

@LuisFelipeCoelho can you look into these points?

**Comment by LuisFelipeCoelho** [comment]
@paulgessinger if I revert it back to TVectorF it seems to work

**Comment by AJPfleger** [comment]
Maybe the uproot problem is solved with the new version? We could try again after #2770

**Comment by LuisFelipeCoelho** [comment]
I think this is good to go. @paulgessinger

**Review by andiwand — APPROVED** [review]
lgtm

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [c4f5a414446b9268da5749ae9723e8199cf54bf5]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840328)||
|:yellow_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840327)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840326)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840325)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840324)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840323)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840322)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840321)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840320)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840319)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840318)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840317)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840316)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840315)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840314)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840313)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840312)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840311)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840310)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840309)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840308)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840307)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840306)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35840305)||

**Pull request merged**